src/ExportHistory/Orchestrations/ExportJobOrchestrator.cs

```diff
@@ -0,0 +1,376 @@
+    {
+        ILogger logger = context.CreateReplaySafeLogger<ExportJobOrchestrator>();
+        string jobId = input.JobEntityId.Key;
+        logger.ExportJobOperationInfo(jobId, nameof(ExportJobOrchestrator), "Export orchestrator started");
+
+        try
+        {
+            // Get the export job state and configuration from the entity
+            ExportJobState? jobState = await context.Entities.CallEntityAsync<ExportJobState?>(
+                input.JobEntityId,
+                nameof(ExportJob.Get),
+                null);
+
+            if (jobState == null || jobState.Config == null)
+            {
+                throw new InvalidOperationException($"Export job '{jobId}' not found or has no configuration.");
+            }
+
+            // Check if job is still active
+            if (jobState.Status != ExportJobStatus.Active)
+            {
+                logger.ExportJobOperationWarning(jobId, nameof(ExportJobOrchestrator), $"Job status is {jobState.Status}, not Active - orchestrator cancelled");
+                return null;
+            }
+
+            ExportJobConfiguration config = jobState.Config;
+
+            int processedCycles = input.ProcessedCycles;
+
+            while (true)
+            {
+                processedCycles++;
+                if (processedCycles > ContinueAsNewFrequency)
+                {
+                    context.ContinueAsNew(new ExportJobRunRequest(input.JobEntityId, ProcessedCycles: 0));
+                    return null!;
+                }
+
+                // Check if job is still active (entity might have been deleted or failed)
+                ExportJobState? currentState = await context.Entities.CallEntityAsync<ExportJobState?>(
+                    input.JobEntityId,
+                    nameof(ExportJob.Get),
+                    null);
+
+                if (currentState == null ||
+                    currentState.Config == null ||
+                    currentState.Status != ExportJobStatus.Active)
+                {
+                    logger.ExportJobOperationWarning(jobId, nameof(ExportJobOrchestrator), "Job is no longer active");
+                    return null;
+                }
+
+                // Call activity to list terminal instances with only necessary information
+                ListTerminalInstancesRequest listRequest = new ListTerminalInstancesRequest(
+                    CompletedTimeFrom: currentState.Config.Filter.CompletedTimeFrom,
+                    CompletedTimeTo: currentState.Config.Filter.CompletedTimeTo,
+                    RuntimeStatus: currentState.Config.Filter.RuntimeStatus,
+                    LastInstanceKey: currentState.Checkpoint?.LastInstanceKey,
+                    MaxInstancesPerBatch: currentState.Config.MaxInstancesPerBatch);
+
+                InstancePage pageResult = await context.CallActivityAsync<InstancePage>(
```

Comment:
Late to the party, but is this why we do `ListInstancesById` instead of `GetAllInstancesAsync`? Because we want to do manual paging? If so, you can disregard my previous comment asking why we do it :)

Reply:
thats true you get me